### ISSUE TITLE
Email not send after upgrade to Magento 2.3.3

### ISSUE THREAD

**Issue by Thomas-gt3** [magento/magento2#25338]
### Preconditions (*)

1. Magento 2.3.3
2. PHP 7.2
3. PLESK 18 

### Steps to reproduce (*)

1. Since update 2.2.8 --> 2.3.3 email not send automatically and not send manually.
2. I tried several modules SMTP and SendInBlue module, but the problem is always present.

### Expected result (*)

1. All Email Send (order, invoice, etc...)

### Actual result (*)

1.
![1](https://user-images.githubusercontent.com/57117461/67725888-39daa180-f9e4-11e9-91f3-cf9d919947a3.png)

2.
![2](https://user-images.githubusercontent.com/57117461/67725893-3e9f5580-f9e4-11e9-91fd-f294eda32aa0.png)

3. While Email is not disabled :

![3](https://user-images.githubusercontent.com/57117461/67725937-642c5f00-f9e4-11e9-8b64-e6a680bb0029.png)

**Comment by ihor-sviziev** [comment]
@Thomas-gt3 there was included backward incompatible change. Please apply the patch listed in release notes for fixing it

https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-3-open-source.html#apply-the-emailmessageinterface-backward-compatibility-issue-patch-to-resolve-an-email-interface-backward-incompatibility-issue

**Comment by Thomas-gt3** [comment]
Thanks, I download patch on my server and when I run it it does not find the files yet I paste the file path :

```
$ cd httpdocs/
$ ls
app		  cgi-bin	 composer.lock	dev										     generated		       img	   lib		    nginx.conf.sample	 pub	      test    vendor
auth.json.sample  CHANGELOG.md	 COPYING.txt	EmailMessageInterface_2.3.3_backward_compatibility_github-2019-10-14-03-45-36.patch  grunt-config.json.sample  index.html  LICENSE_AFL.txt  package.json.sample  SECURITY.md  update
bin		  composer.json  css		favicon.ico									     Gruntfile.js.sample       index.php   LICENSE.txt	    phpserver		 setup	      var
$ patch <EmailMessageInterface_2.3.3_backward_compatibility_github-2019-10-14-03-45-36.patch
can't find file to patch at input line 5
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/app/code/Magento/Email/Model/Transport.php b/app/code/Magento/Email/Model/Transport.php
|index cbce1682cb5..79ceb56a883 100644
|--- a/app/code/Magento/Email/Model/Transport.php
|+++ b/app/code/Magento/Email/Model/Transport.php
--------------------------
File to patch: /app/code/Magento/Email/Model/Transport.php                                                                                        
/app/code/Magento/Email/Model/Transport.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 32
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php b/dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php
|index 32f12dada57..4210312c55a 100644
|--- a/dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php
|+++ b/dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php
--------------------------
File to patch: /dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php
/dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 45
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php b/dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php
|index 10a54b4e1b8..186c3e2796c 100644
|--- a/dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php
|+++ b/dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php
--------------------------
File to patch: /dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php
/dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
3 out of 3 hunks ignored
can't find file to patch at input line 114
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/EmailMessage.php b/lib/internal/Magento/Framework/Mail/EmailMessage.php
|index aaef9750751..02c75977cd0 100644
|--- a/lib/internal/Magento/Framework/Mail/EmailMessage.php
|+++ b/lib/internal/Magento/Framework/Mail/EmailMessage.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/EmailMessage.php
/lib/internal/Magento/Framework/Mail/EmailMessage.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
13 out of 13 hunks ignored
can't find file to patch at input line 308
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php b/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
|index 95f83ff679c..93eaa4acde3 100644
|--- a/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
|+++ b/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 338
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/MailMessageInterface.php b/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
|index da010be2702..5179e6057c4 100644
|--- a/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
|+++ b/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/MailMessageInterface.php
/lib/internal/Magento/Framework/Mail/MailMessageInterface.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 351
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/Message.php b/lib/internal/Magento/Framework/Mail/Message.php
|index b15b75ca9ac..1f423e80108 100644
|--- a/lib/internal/Magento/Framework/Mail/Message.php
|+++ b/lib/internal/Magento/Framework/Mail/Message.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/Message.php
/lib/internal/Magento/Framework/Mail/Message.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 373
Perhaps you should have used the -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php b/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
|index 4a8d6572faa..2d28e2f560a 100644
|--- a/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
|+++ b/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored

```

**Comment by ihor-sviziev** [comment]
Try to use `patch -p1` Instead of `patch`

**Comment by Thomas-gt3** [comment]
Result is identical :

```
$ ls
app		  cgi-bin	 composer.lock	dev										     generated		       img	   lib		    nginx.conf.sample	 pub	      test    vendor
auth.json.sample  CHANGELOG.md	 COPYING.txt	EmailMessageInterface_2.3.3_backward_compatibility_github-2019-10-14-03-45-36.patch  grunt-config.json.sample  index.html  LICENSE_AFL.txt  package.json.sample  SECURITY.md  update
bin		  composer.json  css		favicon.ico									     Gruntfile.js.sample       index.php   LICENSE.txt	    phpserver		 setup	      var
$ patch -p1 < EmailMessageInterface_2.3.3_backward_compatibility_github-2019-10-14-03-45-36.patch
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/app/code/Magento/Email/Model/Transport.php b/app/code/Magento/Email/Model/Transport.php
|index cbce1682cb5..79ceb56a883 100644
|--- a/app/code/Magento/Email/Model/Transport.php
|+++ b/app/code/Magento/Email/Model/Transport.php
--------------------------
File to patch: /app/code/Magento/Email/Model/Transport.php
/app/code/Magento/Email/Model/Transport.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
2 out of 2 hunks ignored
patching file dev/tests/integration/testsuite/Magento/Customer/Controller/AccountTest.php
patching file dev/tests/integration/testsuite/Magento/Framework/Mail/EmailMessageTest.php
can't find file to patch at input line 114
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/EmailMessage.php b/lib/internal/Magento/Framework/Mail/EmailMessage.php
|index aaef9750751..02c75977cd0 100644
|--- a/lib/internal/Magento/Framework/Mail/EmailMessage.php
|+++ b/lib/internal/Magento/Framework/Mail/EmailMessage.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/EmailMessage.php
/lib/internal/Magento/Framework/Mail/EmailMessage.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
13 out of 13 hunks ignored
can't find file to patch at input line 308
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php b/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
|index 95f83ff679c..93eaa4acde3 100644
|--- a/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
|+++ b/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/EmailMessageInterface.php
/lib/internal/Magento/Framework/Mail/EmailMessageInterface.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
2 out of 2 hunks ignored
can't find file to patch at input line 338
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/MailMessageInterface.php b/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
|index da010be2702..5179e6057c4 100644
|--- a/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
|+++ b/lib/internal/Magento/Framework/Mail/MailMessageInterface.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/MailMessageInterface.php
/lib/internal/Magento/Framework/Mail/MailMessageInterface.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 351
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/Message.php b/lib/internal/Magento/Framework/Mail/Message.php
|index b15b75ca9ac..1f423e80108 100644
|--- a/lib/internal/Magento/Framework/Mail/Message.php
|+++ b/lib/internal/Magento/Framework/Mail/Message.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/Message.php
/lib/internal/Magento/Framework/Mail/Message.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
can't find file to patch at input line 373
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php b/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
|index 4a8d6572faa..2d28e2f560a 100644
|--- a/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
|+++ b/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
--------------------------
File to patch: /lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php
/lib/internal/Magento/Framework/Mail/Template/TransportBuilder.php: No such file or directory
Skip this patch? [y] y
Skipping patch.
1 out of 1 hunk ignored
$ 

```

**Comment by gondas33** [comment]
@Thomas-gt3 had the same issue. The problem was due to special characters being used in the sender/receiver name. In my case I was using special characters for my sales e-mail name.

**Comment by emirajbbd** [comment]
@Thomas-gt3 Upload the patch files into magento_root/patches
[patches.zip](https://github.com/magento/magento2/files/3796135/patches.zip)
![2019-11-01_0515](https://user-images.githubusercontent.com/63784/67992480-13786880-fc67-11e9-8763-5d6cca11b5fa.png)
Check the attached screenshot and add the 'extra' part to your composer.json
run composer require cweagans/composer-patches

**Comment by Thomas-gt3** [comment]
@emirajbbd 

Problem persists and now when I send email there is a blank page.
```
$ export PATH=/opt/plesk/php/7.1/bin:$PATH
$ composer require cweagans/composer-patches
Using version ^1.6 for cweagans/composer-patches
./composer.json has been updated
Loading composer repositories with package information
Updating dependencies (including require-dev)
Package operations: 1 install, 0 updates, 0 removals
  - Installing cweagans/composer-patches (1.6.7): Downloading (100%)         
Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Writing lock file
Generating autoload files
$ composer -v install
Gathering patches for root package.
Removing package magento/framework so that it can be re-installed and re-patched.
  - Removing magento/framework (102.0.3)
Removing package magento/module-email so that it can be re-installed and re-patched.
  - Removing magento/module-email (101.0.3)
Loading composer repositories with package information
Installing dependencies (including require-dev) from lock file
Warning: The lock file is not up to date with the latest changes in composer.json. You may be getting outdated dependencies. Run update to update them.
Dependency resolution completed in 0.025 seconds
Analyzed 1020 packages to resolve dependencies
Analyzed 6469 rules to resolve dependencies
Package operations: 2 installs, 0 updates, 0 removals
Installs: magento/module-email:101.0.3, magento/framework:102.0.3
Gathering patches for root package.
Gathering patches for dependencies. This might take a minute.
Found 1 patches for magento/module-email.
Found 1 patches for magento/framework.
  - Installing magento/module-email (101.0.3): Downloading (100%)         
 Extracting archive  - Applying patches for magento/module-email
    patches/EmailMessageInterface_2.3.3_backward_compatibility_module-email.patch (2.3.3 EmailMessageInterface backward compatibility issue patch)
patch '-p1' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/module-email' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_module-email.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/module-email/Model/Transport.php b/vendor/magento/module-email/Model/Transport.php
|index cbce1682cb5..79ceb56a883 100644
|--- a/vendor/magento/module-email/Model/Transport.php
|+++ b/vendor/magento/module-email/Model/Transport.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

patch unexpectedly ends in middle of line
2 out of 2 hunks ignored

patch '-p0' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/module-email' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_module-email.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/module-email/Model/Transport.php b/vendor/magento/module-email/Model/Transport.php
|index cbce1682cb5..79ceb56a883 100644
|--- a/vendor/magento/module-email/Model/Transport.php
|+++ b/vendor/magento/module-email/Model/Transport.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

patch unexpectedly ends in middle of line
2 out of 2 hunks ignored

patch '-p2' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/module-email' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_module-email.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/module-email/Model/Transport.php b/vendor/magento/module-email/Model/Transport.php
|index cbce1682cb5..79ceb56a883 100644
|--- a/vendor/magento/module-email/Model/Transport.php
|+++ b/vendor/magento/module-email/Model/Transport.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

patch unexpectedly ends in middle of line
2 out of 2 hunks ignored

patch '-p4' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/module-email' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_module-email.patch'
(Stripping trailing CRs from patch; use --binary to disable.)

patching file Model/Transport.php
patch unexpectedly ends in middle of line
Hunk #2 succeeded at 61 with fuzz 1.


Patches already gathered. Skipping
  - Installing magento/framework (102.0.3): Downloading (100%)         
 Extracting archive  - Applying patches for magento/framework
    patches/EmailMessageInterface_2.3.3_backward_compatibility_framework.patch (2.3.3 EmailMessageInterface backward compatibility issue patch)
patch '-p1' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/framework' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_framework.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/EmailMessage.php b/vendor/magento/framework/Mail/EmailMessage.php
|index aaef9750751..02c75977cd0 100644
|--- a/vendor/magento/framework/Mail/EmailMessage.php
|+++ b/vendor/magento/framework/Mail/EmailMessage.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

13 out of 13 hunks ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 199
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------

|diff --git a/vendor/magento/framework/Mail/EmailMessageInterface.php b/vendor/magento/framework/Mail/EmailMessageInterface.php
|index 95f83ff679c..93eaa4acde3 100644
|--- a/vendor/magento/framework/Mail/EmailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/EmailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

2 out of 2 hunks ignored
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 229
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/MailMessageInterface.php b/vendor/magento/framework/Mail/MailMessageInterface.php
|index da010be2702..5179e6057c4 100644
|--- a/vendor/magento/framework/Mail/MailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/MailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 242
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Message.php b/vendor/magento/framework/Mail/Message.php
|index b15b75ca9ac..1f423e80108 100644
|--- a/vendor/magento/framework/Mail/Message.php
|+++ b/vendor/magento/framework/Mail/Message.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 264
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Template/TransportBuilder.php b/vendor/magento/framework/Mail/Template/TransportBuilder.php
|index 4a8d6572faa..2d28e2f560a 100644
|--- a/vendor/magento/framework/Mail/Template/TransportBuilder.php
|+++ b/vendor/magento/framework/Mail/Template/TransportBuilder.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

patch '-p0' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/framework' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_framework.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/EmailMessage.php b/vendor/magento/framework/Mail/EmailMessage.php
|index aaef9750751..02c75977cd0 100644
|--- a/vendor/magento/framework/Mail/EmailMessage.php
|+++ b/vendor/magento/framework/Mail/EmailMessage.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

13 out of 13 hunks ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 199
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/EmailMessageInterface.php b/vendor/magento/framework/Mail/EmailMessageInterface.php
|index 95f83ff679c..93eaa4acde3 100644
|--- a/vendor/magento/framework/Mail/EmailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/EmailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

2 out of 2 hunks ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 229
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/MailMessageInterface.php b/vendor/magento/framework/Mail/MailMessageInterface.php
|index da010be2702..5179e6057c4 100644
|--- a/vendor/magento/framework/Mail/MailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/MailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 242
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Message.php b/vendor/magento/framework/Mail/Message.php
|index b15b75ca9ac..1f423e80108 100644
|--- a/vendor/magento/framework/Mail/Message.php
|+++ b/vendor/magento/framework/Mail/Message.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 264
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Template/TransportBuilder.php b/vendor/magento/framework/Mail/Template/TransportBuilder.php
|index 4a8d6572faa..2d28e2f560a 100644
|--- a/vendor/magento/framework/Mail/Template/TransportBuilder.php
|+++ b/vendor/magento/framework/Mail/Template/TransportBuilder.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

patch '-p2' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/framework' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_framework.patch'
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 5
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/EmailMessage.php b/vendor/magento/framework/Mail/EmailMessage.php
|index aaef9750751..02c75977cd0 100644
|--- a/vendor/magento/framework/Mail/EmailMessage.php
|+++ b/vendor/magento/framework/Mail/EmailMessage.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

13 out of 13 hunks ignored
(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 199
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/EmailMessageInterface.php b/vendor/magento/framework/Mail/EmailMessageInterface.php
|index 95f83ff679c..93eaa4acde3 100644
|--- a/vendor/magento/framework/Mail/EmailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/EmailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

2 out of 2 hunks ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 229
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/MailMessageInterface.php b/vendor/magento/framework/Mail/MailMessageInterface.php
|index da010be2702..5179e6057c4 100644
|--- a/vendor/magento/framework/Mail/MailMessageInterface.php
|+++ b/vendor/magento/framework/Mail/MailMessageInterface.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 242
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Message.php b/vendor/magento/framework/Mail/Message.php
|index b15b75ca9ac..1f423e80108 100644
|--- a/vendor/magento/framework/Mail/Message.php
|+++ b/vendor/magento/framework/Mail/Message.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

(Stripping trailing CRs from patch; use --binary to disable.)
can't find file to patch at input line 264
Perhaps you used the wrong -p or --strip option?
The text leading up to this was:
--------------------------
|diff --git a/vendor/magento/framework/Mail/Template/TransportBuilder.php b/vendor/magento/framework/Mail/Template/TransportBuilder.php
|index 4a8d6572faa..2d28e2f560a 100644
|--- a/vendor/magento/framework/Mail/Template/TransportBuilder.php
|+++ b/vendor/magento/framework/Mail/Template/TransportBuilder.php
--------------------------
File to patch: 
Skip this patch? [y] 
Skipping patch.

1 out of 1 hunk ignored

patch '-p4' --no-backup-if-mismatch -d '/var/www/vhosts/********.fr/httpdocs/vendor/magento/framework' < '/var/www/vhosts/********.fr/httpdocs/patches/EmailMessageInterface_2.3.3_backward_compatibility_framework.patch'
(Stripping trailing CRs from patch; use --binary to disable.)

patching file Mail/EmailMessage.php

(Stripping trailing CRs from patch; use --binary to disable.)
patching file Mail/EmailMessageInterface.php

(Stripping trailing CRs from patch; use --binary to disable.)

patching file Mail/MailMessageInterface.php

(Stripping trailing CRs from patch; use --binary to disable.)

patching file Mail/Message.php

(Stripping trailing CRs from patch; use --binary to disable.)

patching file Mail/Template/TransportBuilder.php

Hunk #1 succeeded at 88 (offset 1 line).


Package phpunit/phpunit-mock-objects is abandoned, you should avoid using it. No replacement was suggested.
Generating autoload files
$ 
```

**Comment by Thomas-gt3** [comment]
I restored backup and I think I will wait update Magento 2.3.4

**Comment by DanielRuf** [comment]
A white page typically means that there is some PHP error which you can find in your logfiles.

Does this occur on the current develop branch too?

**Comment by engcom-Delta** [comment]
Hi @Thomas-gt3 thank you for your report. I am not able to reproduce issue by upgrading from 2.2 to 2.3-develop git version and from 2.2.10 to 2.3.3(latest release).
This is issue should be fixed in 2.3.3.
In addition there are patches in [2.3.3 Release Notes](https://devdocs.magento.com/guides/v2.3/release-notes/release-notes-2-3-3-open-source.html#apply-the-emailmessageinterface-backward-compatibility-issue-patch-to-resolve-an-email-interface-backward-incompatibility-issue)

If you'd like to update the issue, please reopen it.

**Comment by jalung** [comment]
I suspect the issue behaves differently depending on the platform being used. Mail is **not** being sent as described by the submitter (from the order review screen and when an order is placed).

In tracking this down:

It is related to the "**-f**" argument that is sent to sendmail via PHP.

Magento supplies 'john@example.com' to Zend Sendmail.php (with the quotes).
Zend prepends **-f** resulting in **-f'john@example.com'**

if PHP php.ini mail setting is as below;

`[mail function]
sendmail_path = /usr/sbin/sendmail -t -i``

then PHP expects **-fjohn@example.com** (without the quotes) or '-fjohn@example.com' with the quotes

if PHP mail setting is as this;
`[mail function]
sendmail_path = /usr/sbin/sendmail -t -i -f
`
then php expects john@example.com (without the quotes and without the -f)

The Temp fix was to preg_replace() the -f in the $parameters of Sendmail.php before the mail() execute function .

Effectively converting -f'john@example.com' to  'john@example.com'
for PHP/sendmail in this site with below config.
`[mail function]
sendmail_path = /usr/sbin/sendmail -t -i -f
`

Quick Example Fix:
Reference class **vendor/zendframework/zend-mail/src/Transport/Sendmail.php** . Fixing in magento code is more involved as the **$parameters** is in the contructor of the magento extended class and mail is created from different areas of the store using overloaded versions of the class.

    public function mailHandler($to, $subject, $message, $headers, $parameters)
    {
        //replace the -f
        if (preg_match('/^ -f/',$parameters) == 1) {
            $parameters = preg_replace('/^ -f/',' ',$parameters);
        }


In summary, these formats 'john@example.com' , '-fjohn@example.com' and -f'john@example.com' apparently work on some platforms, but this Linux expects -fjohn@example.com or 'john@example.com'

**Comment by xtremevision** [comment]
Hi,

We have the same problem, and applying the patch + upgrading to 2.3.4 fixed email sending. However, the new account registration email is not being sent still.

Any ideas?

Thanks,
Michael Mussulis

**Comment by DanielRuf** [comment]
> However, the new account registration email is not being sent still.

Please open a new issue so we can check this in detail.
